### PR TITLE
sync: Improve log for the snapshot purge call

### PR DESCRIPTION
**Pull request by shuo-wu**
Longhorn/longhorn#4294

### CODE REVIEW

**Review by innobead — APPROVED** [review]

**Comment by shuo-wu** [comment]
Will update the instance manager vendor after the PR merge

File: pkg/sync/sync.go [hunk 1]
```diff
@@ -144,19 +144,19 @@ func (t *Task) DeleteSnapshot(snapshot string) error {
 func (t *Task) PurgeSnapshots(skip bool) error {
 	replicas, err := t.client.ReplicaList()
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "failed to list replicas before purging")
```
Comment (innobead) [inline pkg/sync/sync.go:147]: Use `errors.Wrap` instead.

File: pkg/sync/sync.go [hunk 2]
```diff
@@ -179,13 +179,13 @@ func (t *Task) PurgeSnapshots(skip bool) error {
 
 			repClient, err := replicaClient.NewReplicaClient(rep.Address)
 			if err != nil {
-				errorMap.Store(rep.Address, err)
+				errorMap.Store(rep.Address, errors.Wrapf(err, "failed to get replica client %v during purging", rep.Address))
```
Comment (innobead) [inline pkg/sync/sync.go:182]: Suggest "failed to get replica client %v before purging"

**Review by innobead — CHANGES_REQUESTED** [review]

**Comment by innobead** [comment]
@shuo-wu the backport one has been fixed, but this did not. Please help update later, then it's ready to merge.

**Comment by shuo-wu** [comment]
Oh... I didn't find that `git push` was failed

**Review by innobead — APPROVED** [review]

**Pull request merged**